File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int cnt=0;
        int el;
        for (int i=0;i<n;i++){
            if (cnt==0){
                cnt=1;
                el=nums[i];
            }
            else if (el==nums[i]){
                cnt++;
            }
            else{
                cnt--;
            }
        }
        int c=0;
        for (int i=0;i<n;i++){
            if (nums[i]==el){
                c++;
            }
        }
        if (c>(n/2)){
            return el;
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] arr) {
        int ele=0;
        int cnt=0;
        int n=arr.length;
        for(int i=0;i<n;i++){
            if(cnt==0){
                ele=arr[i];
            }
            if(ele==arr[i]) cnt++;
            else cnt--;
        }
        return ele;
    }
}